internal/service/auth.go

```diff
@@ -0,0 +1,62 @@
+package service
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/avialog/backend/internal/dto"
+	"github.com/avialog/backend/internal/infrastructure"
+	"github.com/avialog/backend/internal/model"
+	"github.com/avialog/backend/internal/repository"
+)
+
+//go:generate mockgen -source=auth.go -destination=auth_mock.go -package service
+type AuthService interface {
+	ValidateToken(ctx context.Context, token string) (model.User, error)
+}
+
+type authService struct {
+	userRepository repository.UserRepository
+	authClient     infrastructure.AuthClient
+}
+
+func newAuthService(userRepository repository.UserRepository, authClient infrastructure.AuthClient) AuthService {
+	return &authService{userRepository: userRepository, authClient: authClient}
+}
+
+func (a *authService) ValidateToken(ctx context.Context, token string) (model.User, error) {
+	var newUser model.User
+
+	response, err := a.authClient.VerifyIDToken(ctx, token)
+	if err != nil {
+		return model.User{}, fmt.Errorf("%w: %v", dto.ErrInternalFailure, err)
+	}
+	userEmail := response.Claims["email"].(string)
```

Comment:
Please add check if `map[string]interface{}` does not contain key such as `email`. Also add check if `response.Claims["email"]` is actually `string`, as invalid type cast causes panics in Go